netty/src/main/java/io/grpc/netty/NettyClientHandler.java

```diff
@@ -340,7 +340,7 @@ private void createStream(CreateStreamCommand command, final ChannelPromise prom
       if (!connection().goAwaySent()) {
         logger.fine("Stream IDs have been exhausted for this connection. "
                 + "Initiating graceful shutdown of the connection.");
-        super.close(ctx(), ctx().newPromise());
+        close(ctx(), ctx().newPromise());
```

Comment:
Hmm... looking at gracefulClose makes me think I should actually call lifecycleManager.notifyShutdown here to provide the correct status. Then it wouldn't matter too much whether we called super or this.close()